src/main/java/com/scriptbasic/executors/commands/CommandCase.java

```diff
@@ -0,0 +1,143 @@
+			final var operator = new RightSideEqualsOperator(selectExpressionValue);			
+			operator.setRightOperand(expression);
+			final var conditionValue = operator.evaluate(interpreter);
+	        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+	            return BasicBooleanValue.asBoolean(conditionValue);
+	        } else {
+	            throw new BasicRuntimeException(
+	                    "Case condition can not be evaluated to boolean");
+	        }
+		}
+	}
+	
+	public static class FromToCaseCondition implements CaseCondition {
+		final private Expression fromExpression;
+		final private Expression toExpression;
+
+		public FromToCaseCondition(final Expression fromExpression, final Expression toExpression) {
+			this.fromExpression = fromExpression;
+			this.toExpression = toExpression;
+		}
+
+		@Override
+		public boolean matchCase(Interpreter interpreter, RightValue selectExpressionValue) 
+				throws ScriptBasicException {
+			final var leftOperator = new RightSideGreaterOrEqualOperator(selectExpressionValue);
+			leftOperator.setRightOperand(fromExpression);
+			var conditionValue = leftOperator.evaluate(interpreter);
+	        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+	            if(!BasicBooleanValue.asBoolean(conditionValue)) {
+	            	return false;
+	            }
+	            // evaluate to condition
+	            final var rightOperator = new RightSideLessOrEqualOperator(selectExpressionValue);
+	            rightOperator.setRightOperand(toExpression);
+	            conditionValue = rightOperator.evaluate(interpreter);
+		        if (conditionValue instanceof AbstractPrimitiveRightValue<?>) {
+		        	return BasicBooleanValue.asBoolean(conditionValue);
+		        }
+	        }
+
+	        throw new BasicRuntimeException("Case condition can not be evaluated to boolean");
+		}
+		
+	}
+	
+	/**
+	 * List of case conditions to be evaluated
+	 */
+	private List<CaseCondition> expressionList = new ArrayList<>();
+	private CommandEndSelect commandEndSelect;
+
+	@Override
+	public void execute(Interpreter interpreter) throws ScriptBasicException {
+
+		// Check if other case already applied
+		Boolean caseApplied = (Boolean) interpreter.getMap().get(CommandSelect.CASE_APPLIED);
+		if(caseApplied) {
+			// skip to end
+			interpreter.setNextCommand(commandEndSelect);
+		} else {
+			// set as applied and execute this case
+		    
+			interpreter.getMap().put(CommandSelect.CASE_APPLIED, true);
+			
+			interpreter.setNextCommand(getNextCommand());		
+		}
+		
+	}
+
+	public void addCaseCondition(final CaseCondition caseCondition) {
+		expressionList.add(caseCondition);
+	}
+
+	public boolean matches(Interpreter interpreter, RightValue expressionValue) throws ScriptBasicException { 
+	    // default case has empty list of expressions
+		if(expressionList.size()==0) {
+			return true;
+		}
+		
+		// check case specific conditions

```

Comment:
this is a typical logic that can be expressed using Java 8 streams. Something like

```
expressionList.stream().anyMatch(caseCondition -> caseCondition.matchCase(interpreter, expressionValue))
```

Reply:
Streams cannot be used here (or I do not know how) because caseCondition.matchCase throws Exceptions